src/frontend/screens/MapScreen/track/saveTrack/SaveTrackButton.tsx

```diff
@@ -40,7 +36,7 @@ export const SaveTrackButton: FC<SaveTrackButton> = ({description}) => {
       },
       {
         onSuccess: () => {
-          navigation.navigate(TabName.Map);
+          navigation.navigate('Home', {screen: 'Map'});
```

Comment:
We actually need to reset the stack when we navigate to the map. 

So normally when we navigate to the 'map', the map is the first thing in the stack. React navigation handles this by going `back` to the map, instead of adding it to the stack => this mean after a user saves a track, they are navigated `back` to the map. And if they were to press the back button, then they would not be able to go back to the tracks screen

With the persisted state, the tracks screen is sometimes the first screen (because of setting the initial route name). So if that is the case, maps is not part of the stack, and when they are navigated to maps, it is added to the stack. So the user is able to press the back button and go to the tracks screen, which is undesired behaviour.

In order to avoid this behavior we need to [reset the navigation](https://reactnavigation.org/docs/navigation-actions/#reset) to make maps the only thing in the stack.